### PR TITLE
3.2 is done

### PR DESCRIPTION
**Pull request by dmytro-hryhoruk**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -16,6 +16,33 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public int hashCode() {
+        int result = 0;
+        result = color.length() + width.intValue() + length.intValue();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]: is it ok to use length of fields to find hashcode?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -16,6 +16,33 @@ public Rectangle(Integer width, Integer length, String color) {
...
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java:27]: how about checking on null?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -16,6 +16,33 @@ public Rectangle(Integer width, Integer length, String color) {
...
+        if (this == obj) {
+            return true;
+        }
+        if (obj instanceof Rectangle) {
+            Rectangle rectangleobj = (Rectangle) obj;
+            if (!this.color.equals(rectangleobj.color)) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]: can we simplify this three checking using && or || ?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 4]
```diff
@@ -16,6 +16,33 @@ public Rectangle(Integer width, Integer length, String color) {
...
+        result = color.length() + width.intValue() + length.intValue();
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj instanceof Rectangle) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]: what is the difference between `instanceof` and `getClass()`?

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Rectangle.java]: Actually I don't know for sure but it has worked in this video https://youtu.be/Lk2vF7xz8tE?t=667

Comment (Sofasmile) [reply src/main/java/core/basesyntax/Rectangle.java]: which hash code we will have if we create next objects:
Rectangle rec1 = new Rectangle(20, 25, "green");
Rectangle rec2 = new Rectangle(20, 25, "black");
rec1.hachCode()   ?   rec2.hachCode() 

our hachCode() shoud be different

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Rectangle.java]: after Q&A session somehow my latest commit didn't get to GitHub. Now that's gonna be ok.

File: src/main/java/core/basesyntax/Rectangle.java [hunk 5]
```diff
@@ -16,14 +16,35 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
-    public Integer getWidth() {
-        return width;
+    @Override
+    public int hashCode() {
+        int result = 0;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java:21]:
```suggestion
        int result = 31;
```
we should use some primary number)

File: src/main/java/core/basesyntax/Rectangle.java [hunk 6]
```diff
@@ -16,14 +16,35 @@ public Rectangle(Integer width, Integer length, String color) {
...
+        result = color.hashCode() + width.hashCode() + length.hashCode();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
        result += color.hashCode() + width.hashCode() + length.hashCode();
```

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**